src/choose-location/choose-location.component.tsx

```diff
@@ -48,160 +44,47 @@ export default function ChooseLocation(props: ChooseLocationProps) {
           }
         })
         .catch(createErrorHandler());
-    }
-    return () => abortController.abort();
-  }, [isSubmitting]);
+    },
+    [referrer]
+  );
+
   React.useEffect(() => {
     const ac = new AbortController();
-    changeLocationData({ locationResult: props.loginLocations });
-    if (props.loginLocations.length > 100) {
-      if (searchTerm) {
-        searchLocationsFhir(searchTerm).then((locs) => {
-          changeLocationData({ locationResult: locs.data.entry });
-          if (isEmpty(locs.data.entry)) {
-            changeLocationData({ emptyResult: true });
-          } else {
-            changeLocationData({ emptyResult: false });
-          }
-        }, createErrorHandler());
-      }
-    } else if (searchTerm) {
-      filterList(searchTerm);
-    } else {
-      changeLocationData({ emptyResult: false });
-      changeLocationData({ locationResult: props.loginLocations });
-    }
-    return () => ac.abort();
-  }, [searchTerm, props.loginLocations]);
+    const sub = getCurrentUser().subscribe((user) => {
+      setCurrentUser(user ? user.display : currentUser);
+    }, createErrorHandler());
 
-  const search = debounce((location) => {
-    setSearchTerm(location);
-  }, searchTimeout);
+    queryLocations("", ac).then(
+      (locations) => setLoginLocations(locations),
+      createErrorHandler()
+    );
 
-  const filterList = (searchTerm) => {
-    if (searchTerm) {
-      const updatedList = locationData.locationResult.filter((item) => {
-        return (
-          item.resource.name.toLowerCase().search(searchTerm.toLowerCase()) !==
-          -1
-        );
-      });
-      if (updatedList.length > 0) {
-        changeLocationData({ locationResult: updatedList });
+    return () => {
+      ac.abort();
+      sub.unsubscribe();
+    };
+  }, []);
+
+  React.useEffect(() => {
+    if (loginLocations && currentUser) {
+      if (!config.chooseLocation.enabled || loginLocations.length < 2) {
+        changeLocation(loginLocations[0]?.resource.id);
```

Comment:
Nit: superfluous optional chaining

Reply:
Why? If we have 0 `loginLocations` this will crash. The optional chaining is thus required to make it work even in the 0 length case.

Reply:
Whoops, I must have been a little tired, I misread the `if` block there. All good.